src/main/java/com/coreos/jetcd/lease/LeaseKeepAliveResponseWithError.java

```diff
@@ -0,0 +1,20 @@
+package com.coreos.jetcd.lease;
+
+import com.coreos.jetcd.api.LeaseKeepAliveResponse;
+
+/**
+ * Created by fanminshi on 6/8/17.
+ */
+public class LeaseKeepAliveResponseWithError {
+
+  public LeaseKeepAliveResponse leaseKeepAliveResponse;
+  public Exception error;
```

Comment:
it is a poison pill that is used to notify the consumer thread something is wrong.
```java
@Override
    public synchronized LeaseKeepAliveResponse listen()
        throws InterruptedException {
...
      Future<LeaseKeepAliveResponse> future = service.submit(() -> {
        LeaseKeepAliveResponseWithError lkae = this.queue.take();
        if (lkae.error != null) {
          this.reason = lkae.error;
          throw lkae.error; // throw here. 
        }
        return lkae.leaseKeepAliveResponse;
      });
...
```